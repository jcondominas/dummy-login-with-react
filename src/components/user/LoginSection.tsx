import * as React from "react";
import './loginSection.css'

export interface LogginSectionProps {
    onLogin: (user: string, password: string) => any
}

interface LoginSectionState {
    user: string,
    password: string
}

export class LoginSection extends React.Component<LogginSectionProps, LoginSectionState> {
    constructor(props: LogginSectionProps) {
        super(props);
        this.state = {user: "", password: ""}
    }

    public render() {
        return (
            <form onSubmit={(event) => {
                event.preventDefault();
                this.props.onLogin(this.state.user, this.state.password);
            }}
            className="form-login">
                <input type="text" placeholder="Username" onChange={(event) => {
                    this.setState({
                        user: event.target.value
                    })
                }}/>
                <input type="text" placeholder="Password" onChange={(event) => {
                    this.setState({
                        password: event.target.value
                    })
                }}/>
                <input type="submit"/>
            </form>
        )
    }

}