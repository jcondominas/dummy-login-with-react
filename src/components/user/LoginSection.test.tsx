import {LogginSectionProps, LoginSection} from "./LoginSection";
import {shallow} from "enzyme";
import * as React from "react";
import toJson from 'enzyme-to-json'
import Mock = jest.Mock;

describe('LoginSection', () => {
    it('should send log in action when form submited', () => {
        const LoginMock = jest.fn<LogginSectionProps>(() => ({
            onLogin: jest.fn()
        }));
        const props = new LoginMock();
        const loginSection = shallow(<LoginSection onLogin={props.onLogin}/>);

        loginSection.find("input").first().simulate("change", {
            target: {
                value: "UserName"
            }
        });
        loginSection.find("input").at(1).simulate("change", {
            target: {
                value: "Password"
            }
        });
        loginSection.find("form").simulate("submit", {preventDefault: jest.fn});

        expect((props.onLogin as Mock).mock.calls[0]).toEqual(['UserName', 'Password'])
    });

    it('should renders login section correctly', () => {
        const login = shallow(<LoginSection onLogin={jest.fn()}/>);
        expect(toJson(login)).toMatchSnapshot()
    });
});