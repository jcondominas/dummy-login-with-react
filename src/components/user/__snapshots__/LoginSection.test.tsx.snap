// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginSection should renders login section correctly 1`] = `
<form
  className="form-login"
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Username"
    type="text"
  />
  <input
    onChange={[Function]}
    placeholder="Password"
    type="text"
  />
  <input
    type="submit"
  />
</form>
`;
